### PR TITLE
DOC: Fixed type hints for GroupBy.{any, all}

### PR DESCRIPTION
**Pull request by ytakashina**
The current documentation says the return value of GroupBy.all and GroupBy.any is `bool`.

```
    def any(self, skipna: bool = True):
        """
        Return True if any value in the group is truthful, else False.

        Parameters
        ----------
        skipna : bool, default True
            Flag to ignore nan values during truth testing.

        Returns
        -------
        bool
        """
```

However, the actual returned type is DataFrame or Series (with the same shape as `GroupBy.sum()`).

```
In [25]: df
Out[25]:
     0    1    2    3
0  1.0  0.0  0.0  0.0
1  0.0  1.0  0.0  0.0
2  0.0  0.0  1.0  0.0
3  0.0  0.0  0.0  1.0

In [26]: df.groupby(0).any()
Out[26]:
         1      2      3
0
0.0   True   True   True
1.0  False  False  False

In [27]: type(df.groupby(0).any())
Out[27]: pandas.core.frame.DataFrame
```

This PR fixes the wrong type hints for `GroupBy.{any, all}`.

- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by phofl** [comment]
That is a bit short imho. Could you maybe write something like here ? https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.GroupBy.sum.html

**Comment by ytakashina** [comment]
Okay. I wrote some drafts for `GroupBy.all()` documentation. Which one do you think better?

1. Computed all of values within each group.
    - just replaced `sum` by `all` in GroupBy.sum documentation.
2. Computed all() of values within each group.
    - added parentheses to make it clear that `all` represents somewhat function (that is, it's not a pronoun).
3. DataFrame or Series of boolean values, which is True if all elements are True within each group, False otherwise.
    - wrote whole description.

If you have any alternative, plz tell me.

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1330,7 +1330,9 @@ def any(self, skipna: bool = True):
 
         Returns
         -------
-        bool
+        Series or DataFrame
+            DataFrame or Series of boolean values, which is True if any element is True
+            within each group, False otherwise.
```
Comment (ytakashina) [inline pandas/core/groupby/groupby.py:1335]: I've pushed the third draft for now. But I'm willing to modify these descriptions if requested.

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -1330,7 +1330,9 @@ def any(self, skipna: bool = True):
 
         Returns
         -------
-        bool
+        Series or DataFrame
+            DataFrame or Series of boolean values, which is True if any element is
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:1334]: What do you think about: "..., where a value is True if any element is True within its respective group..."

File: pandas/core/groupby/groupby.py [hunk 3]
```diff
@@ -1330,7 +1330,9 @@ def any(self, skipna: bool = True):
...
+            True within each group, False otherwise.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:1335]: Should this truthful instead of True?

File: pandas/core/groupby/groupby.py [hunk 4]
```diff
@@ -1347,7 +1349,9 @@ def all(self, skipna: bool = True):
 
         Returns
         -------
-        bool
+        Series or DataFrame
+            DataFrame or Series of boolean values, which is True if all elements are
+            True within each group, False otherwise.
```
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:1354]: Similar to the above.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py (hunk 2 above)
Comment (ytakashina) [reply pandas/core/groupby/groupby.py:1334]: Sounds more precise than the current one. Thank you!

File: pandas/core/groupby/groupby.py (hunk 3 above)
Comment (ytakashina) [reply pandas/core/groupby/groupby.py:1335]: It looks `True` is commonly used in pandas documentation.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.any.html
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.all.html

`True` (capitalized) is not just a noun, but it represents the boolean literal in Python. So I believe `True` is appropriate in this case.

**Comment by ytakashina** [comment]
Edited the description for the return value of GroupBy.{any, all} after receiving @rhshadrach's suggestion.

**Comment by jbrockmendel** [comment]
Seems like a clear improvement over the status quo.  @rhshadrach are you happy here?

File: pandas/core/groupby/groupby.py (hunk 3 above)
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py]: Thanks for checking this - agreed.

**Review by rhshadrach — APPROVED** [review]
Looks great - thanks @ytakashina

**Comment by rhshadrach** [comment]
@ytakashina Can you merge master? I think it will fix CI.

**Comment by ytakashina** [comment]
@rhshadrach CI Fixed!

**Comment by rhshadrach** [comment]
Thanks @ytakashina!

**Pull request merged**